gitpod-io/gitpod
# [kots]: change to the new installer config commands

## Description

Configures the KOTS installer to use the shiny new Installer `config` commands.

The `kots-install.sh` still merges the Docker config JSON files. Ultimately, this will live in the Installer binary, but that work is not yet ready as dependent upon #12060.

This requires #12687 to be merged first

## Related Issue(s)

Fixes #12984

## How to test

Install via KOTS. As this changes the whole installation process, we should confirm that all the normal deployment tests run. When test passed satisfactorily, please check it and put your name:
- [x] AWS @MrSimonEmms 
- [x] Azure @MrSimonEmms 
- [x] Azure (airgap) @MrSimonEmms 
- [x] GCP @MrSimonEmms 
- [x] k3s @MrSimonEmms
- [x] k3s (proxy - the test here is for it to start as [still work to be done](https://github.com/gitpod-io/gitpod/issues/10769)) @MrSimonEmms 
- [x] preview environment @MrSimonEmms 
- [x] werft publish-to-kots @MrSimonEmms 
- [x] [AKS auto-tests](https://werft.gitpod-dev.com/job/gitpod-custom-sje-kots-refactoring.1) n/a - there's an inconsistency in the test. The manual test works, so will ignore @MrSimonEmms 
- [x] [EKS auto-tests](https://werft.gitpod-dev.com/job/gitpod-custom-sje-kots-refactoring.0) n/a - there's an inconsistency in the test. The manual test works, so will ignore @MrSimonEmms 
- [x] [GKE auto-tests](https://werft.gitpod-dev.com/job/gitpod-custom-sje-installer-config-build.7) @MrSimonEmms 
- [x] [k3s auto-tests](https://werft.gitpod-dev.com/job/gitpod-custom-sje-installer-config-build.4) @MrSimonEmms 

## Release Notes

```release-note
[kots]: change to the new installer config commands
```

## Documentation


## Werft options:
